File: 1118.py
med = 0
x = 1
i=0
while x == 1:
    while i < 2:
        n = float(input())
        if n < 0.0 or n > 10.0:
            print('nota invalida')
        if n >= 0.0 and n <=10.0:
            med = med + n
            i = i + 1
    print('media = {:.2f}'.format(med / 2))

    while True:
        x = int(input('novo calculo (1-sim 2-nao)'))
        i =0
        med = 0
        if x == 2 or x == 1:
            break

File: CSharp/1118.cs
using System;
using System.Collections.Generic;
using System.Linq;

class URI
{
    static void Main()
    {
        List<float> nota = new List<float>();
        float x;
        bool True =true;

        while(True)
        {
            while(True)
            {
                x = float.Parse(Console.ReadLine());
                if(x<0 || x>10)
                {
                    Console.WriteLine("nota invalida");
                }
                else
                {
                    nota.Add(x);
                }

                if(nota.Count==2)
                {
                    Console.WriteLine($"media = {nota.Average():F2}");
                    nota.Clear();
                    break;
                }
            }
            while(True)
            {
                Console.WriteLine("novo calculo (1-sim 2-nao)");
                x = float.Parse(Console.ReadLine());
                if(x==2)
                {True=false;break;}
                else if(x==1)
                {break;}
            }
        }
    }
}